File: java/src/org/raml/amf/core/domain/Parameter.java
package org.raml.amf.core.domain;

import api_modeling_framework.model.domain.ParsedParameter;
import org.raml.amf.core.exceptions.InvalidModelException;
import org.raml.amf.utils.Clojure;

/**
 * Created by antoniogarrote on 04/05/2017.
 */
public class Parameter extends GenericParameter {

    static {
        Clojure.require(Clojure.API_MODELING_FRAMEWORK_MODEL_DOMAIN);
    }

    public Parameter(ParsedParameter rawModel) {
        super(rawModel);
    }

    public Parameter(String id, String parameterKind) {
        super((ParsedParameter) Clojure.var(
                Clojure.API_MODELING_FRAMEWORK_MODEL_DOMAIN,
                "api-modeling-framework.model.domain/map->ParsedParameter"
                ).invoke(Clojure.map())
        );
        this.setId(id);
        this.setParameterKindInternal(parameterKind);
    }

    protected void setParameterKind(String parameterKind) {
        setParameterKindInternal(parameterKind);
    }

    public String getParameterKind() {
        Object res =  this.wrapped().parameter_kind();
        if (res != null) {
            return (String) res;
        } else {
            return null;
        }
    }


    protected ParsedParameter wrapped() {
        return (ParsedParameter) this.rawModel;
    }
}

File: bindings/js/src/core/domain/Parameter.ts
/**
 * Created by antoniogarrote on 05/05/2017.
 */

import {Clojure} from "../../Clojure";
import {DomainModel} from "./DomainModel";
import {GenericParameter, ParameterKind} from "./GenericParameter";

/**
 * HTTP parameter other than a header. It can be located in the domain, path or query
 */
export class Parameter extends GenericParameter {

    public static build(id: string, parameterKind: ParameterKind): Parameter {
        return new Parameter(Clojure.amf.build(
            DomainModel.domain_builder,
            Clojure.amf_domain.map__GT_ParsedParameter,
            id));
    }
}